crates/wast/src/component/binary.rs

```diff
@@ -884,15 +880,17 @@ impl From<&ModuleType<'_>> for wasm_encoder::ModuleType {
 
         for decl in &ty.decls {
             match decl {
-                ModuleTypeDecl::Type(t) => match &t.def.kind {
-                    core::InnerTypeKind::Func(f) => encoded.ty().function(
-                        f.params.iter().map(|(_, _, ty)| (*ty).into()),
-                        f.results.iter().copied().map(Into::into),
-                    ),
-                    core::InnerTypeKind::Struct(_) | core::InnerTypeKind::Array(_) => {
-                        todo!("encoding of GC proposal types not yet implemented")
-                    }
-                },
+                ModuleTypeDecl::Type(t) => {
+                    let sub_type = SubType {
+                        is_final: t.final_type.unwrap_or(true), // TODO: is this correct?
+                        supertype_idx: t.parent.map(u32::from), // TODO: resolve?
```

Comment:
For this TODO, do you mean resolving `$t` to an index? If so that happens in prior phases so I think it's ok to remove this TODO